master/buildbot/status/web/status_json.py

```diff
@@ -1371,6 +1376,31 @@ def asDict(self, request):
         defer.returnValue(build_number)
 
 
+class MyBuildsJsonResource(JsonResource):
+    help = 'Gives information about my builds from last 7 days'
+    pageTitle = 'My builds from last 7 days'
+
+    @defer.inlineCallbacks
+    def asDict(self, request):
+        from buildbot.status.web.authz import Authz
+        from buildbot.status.web.mybuilds import MybuildsResource
+        from datetime import date, datetime
+
+        def json_serial(obj):
+            """JSON serializer for objects not serializable by default json code"""
+
+            if isinstance(obj, (datetime, date)):
+                return obj.isoformat()
+            raise TypeError("Type %s not serializable" % type(obj))
+
+        authz = Authz()
+        mybuilds = MybuildsResource()
+        master = mybuilds.getBuildmaster(request)
+        username = authz.getUsernameFull(request)
```

Comment:
It would be nice if username could also come as a parameter, so we can query builds from other users.